security/monitoring.adoc

```diff
@@ -0,0 +1,197 @@
+2017-10-17T13:12:17.636081056Z AUDIT: id="410eda6b-88d4-4491-87ff-394804ca69a1" response="200"
+----
+
+For this example, it may be useful to poll the log periodically using the following
+command to see if the number of failed requests:
+
+----
+$ tail -5000 /var/log/openshift-audit.log \
+  | grep -Po 'response="..."' \
+  | sort | uniq -c | sort -rn
+
+   3288 response="200"
+      8 response="404"
+      6 response="201"
+----
+
+Most valid requests succeed with a 200 or 201 return code. 401 and
+403 are simple redirects, and 404 is typically a benign request for a
+resource that does not exist, but return code 400 may be of
+interest as it indicates a malformed request, which should not occur with
+most clients. Responses in the 500 range indicate server errors, which
+could be a result of bugs, system failures, or even malicious activity.
+If unusual numbers of error responses are found, the audit log entries
+for corresponding requests can be retrieved for further investigation.
+
+It could also be useful to look for unusual numbers of requests by a
+particular user or group. Note that the IP address of the request is typically
+a cluster host or API load balancer, and there is no record of the IP address
+behind a load balancer proxy request (however, load balancer logs could
+potentially be useful for determining request origin). The following
+example lists the top ten users by number of requests in the last 5000
+lines of the audit log:
+
+----
+$ tail -5000 /var/log/openshift-audit.log \
+  | grep -Po ' user="(.*?)(?<!\\)"' \
+  | sort | uniq -c | sort -rn | head -10
+
+    976  user="system:openshift-master"
+    270  user="system:node:origin-node-1.example.local"
+    270  user="system:node:origin-master.example.local"
+     66  user="system:anonymous"
+     32  user="system:serviceaccount:kube-system:cronjob-controller"
+     24  user="system:serviceaccount:kube-system:pod-garbage-collector"
+     18  user="system:serviceaccount:kube-system:endpoint-controller"
+     14  user="system:serviceaccount:openshift-infra:serviceaccount-pull-secrets-controller"
+     11  user="test user"
+      4  user="test \" user"
+----
+
+More sophisticated queries generally require more sophisticated
+log analysis tools. Auditors will also need a detailed familiarity
+with the xref:../rest_api/openshift_v1.adoc#rest-api-openshift-v1:[OpenShift v1 API]
+and xref:../rest_api/kubernetes_v1.adoc#rest-api-kubernetes-v1:[Kubernetes v1 API]
+to aggregate request summaries from the audit log according to which
+kind of resource is involved (the *_uri_* field).
+
+xref:../install_config/master_node_configuration.adoc#master-node-config-advanced-audit[More advanced audit logging capabilities]
+are introduced with {product-title} 3.7 as a technology preview.
```

Comment:
Technology Preview feature

Reply:
Also, it will be a good idea to move this up top in the Introduction section. For example https://docs.openshift.com/container-platform/latest/architecture/service_catalog/index.html#service-catalog-overview

Reply:
Fixed.